account_billing/tests/test_account_billing.py

```diff
@@ -0,0 +1,270 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import Command, fields
+from odoo.exceptions import UserError, ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestAccountBilling(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.journal_sale = cls.env["account.journal"].search([("type", "=", "sale")])[0]
+        cls.invoice_model = cls.env["account.move"]
+        cls.invoice_line_model = cls.env["account.move.line"]
+        cls.billing_model = cls.env["account.billing"]
+        cls.register_payments_model = cls.env["account.payment.register"]
+
+        cls.partner_id = cls.env["res.partner"].create({"name": "Test Partner"})
+        cls.payment_term = cls.env.ref("account.account_payment_term_15days")
+        cls.partner_agrolait = cls.env.ref("base.res_partner_2")
+        cls.partner_china_exp = cls.env.ref("base.res_partner_3")
+        cls.product = cls.env.ref("product.product_product_4")
+        cls.currency_eur = cls.env.ref("base.EUR")
+        cls.currency_eur.active = True
+        cls.currency_usd_id = cls.env.ref("base.USD").id
+        cls.currency_eur_id = cls.currency_eur.id
```

Comment:
Just to be consistent and to avoid having unnecessary objects.
```suggestion
        cls.currency_usd = cls.env.ref("base.USD")
        cls.currency_usd.active = True
```